src/components/ScaleLattice.vue

```diff
@@ -1,19 +1,282 @@
+const props = defineProps<{
   scale: Scale;
   width: number;
   height: number;
   heldScaleDegrees: Set<number>;
   keyColors: string[]; // Use key colors in visualization if you like. Not synced with scale size, though...
 }>();
 
-const canvas = ref<HTMLCanvasElement | null>(null);
+type Coord = { x: number; y: number };
+type Node = {
+  x: number;
+  y: number;
+  label: string;
+  index: number;
+  fill: string;
+};
+
+type Coords = {
+  [key: string]: Coord;
+};
+
+const coords: Coords = {
+  "1": {
+    x: 0,
+    y: 0,
+  },
+  "2": {
+    x: 0,
+    y: 0,
+  },
+  "3": {
+    x: 40,
+    y: 0,
+  },
+  "5": {
+    x: 0,
+    y: -40,
+  },
+  "7": {
+    x: 13,
+    y: -11,
+  },
+  "11": {
+    x: -14,
+    y: -18,
+  },
+  "13": {
+    x: -8,
+    y: -4,
+  },
+  "17": {
+    x: -5,
+    y: -32,
+  },
+  "19": {
+    x: 7,
+    y: -25,
+  },
+  "23": {
+    x: 20,
+    y: -6,
+  },
+};
+
+// === Computed state ===
+const canLatticePrimes = computed(() =>
+  // hardcoded for now to the 23-limit
+  props.scale.intervals.reduce(
+    (canLattice: boolean, interval: Interval) =>
+      canLattice &&
+      interval.monzo
+        .toIntegerMonzo()
+        .slice(9)
+        .reduce(
+          (isZero: boolean, component: number) => isZero && component === 0,
+          true
+        ),
+    true
+  )
+);
+
+function maybeGetPrimeEquave(equave: Interval): number | null {
+  const intmonzo: number[] = equave.monzo.toIntegerMonzo();
+  return intmonzo.reduce(
+    (
+      primeIndex: number | null,
+      component: number,
+      i: number
+    ): number | null => {
+      if (component === 1 && !primeIndex) {
+        return i;
+      }
+      if (component !== 0) {
+        console.error("Equave is not a prime number");
+        return primeIndex;
+      }
+      if (component === 0) {
+        return primeIndex;
+      }
+      return primeIndex;
+    },
+    null
+  );
+}
+
+const nodes: ComputedRef<Node[] | Error> = computed(() => {
+  try {
+    const equavePrimeIndex = maybeGetPrimeEquave(props.scale.equave);
+    return props.scale.intervals.map(({ monzo, name: label }, index) => {
+      const node: Node = monzo.toIntegerMonzo().reduce(
+        (node, component, i): Node => {
+          const { x, y } = node;
+          if (i === equavePrimeIndex) {
+            return node;
+          }
+
+          const componentVal = component.valueOf();
+          if (componentVal === 0) {
+            return node;
+          } else {
+            const p: string = PRIMES[i].toString();
+            const vector = coords[p] || { x: 0, y: 0 };
+            return {
+              ...node,
+              x: x + vector.x * componentVal,
+              y: y + vector.y * componentVal,
+            };
+          }
+        },
+        { x: 0, y: 0, label, index, fill: "white" }
+      );
+      return node;
+    });
+  } catch {
+    return new Error("Cannot make lattice of non JI scale.");
```

Comment:
You `throw` errors, not `return` them.

Reply:
(Maybe this was part of some other machinery. Just came home, didn't look too deep. In any case it's not good practice to repurpose the `Error` class for other uses. Make a similar custom class if you must.)